DREAM/Optimization/tsgOptimizationUtils.hpp

```diff
@@ -57,6 +57,16 @@
 
 namespace TasOptimization {
 
+/*! \internal
+ * \ingroup OptimizationUtil
+ *
+ * Stores information about the run of an optimization algorithm.
+ */
+struct OptimizationStatus {
+    int num_iterations;
+    double stationarity_residual;
```

Comment:
Change this to `residual` and we will explain what it means for each method.

Here we can add a note that the definition of the `residual` can depend on the algorithm.